### PR TITLE
FIX #23 : Add requirement for copyright notice to CONTRIBUTING.md

### LINKED ISSUE #23: CONTRIBUTING.md should mention copyright notice

### PR DESCRIPTION
**Pull request by alin00**

### CODE REVIEW

**Comment by charliegracie** [comment]
Should we include the text for the copyright notice?

**Comment by alin00** [comment]
I was being ambiguous/lazy because I'm not 100% sure what the copyright notice should be for files with multiple copyright owners.

@mstoodle Is this template right?

```c
/*******************************************************************************
 *
 * (c) Copyright <copyright owner #1> <first year written>, <last year modified>
 * (c) Copyright <copyright owner #2> <first year written>, <last year modified>
 *
 *  This program and the accompanying materials are made available
 *  under the terms of the Eclipse Public License v1.0 and
 *  Apache License v2.0 which accompanies this distribution.
 *
 *      The Eclipse Public License is available at
 *      http://www.eclipse.org/legal/epl-v10.html
 *
 *      The Apache License v2.0 is available at
 *      http://www.opensource.org/licenses/apache2.0.php
 *
 * Contributors:
 *    <First author> - initial implementation and documentation
 *******************************************************************************/
```

**Comment by charliegracie** [comment]
Also are we going to get people to add their name and a comment under the Contributors?

**Comment by alin00** [comment]
I thought we had decided not to update the contributor list for our team, since eclipse doesn't absolutely require it. We haven't been doing it for any recent commits.

**Comment by alin00** [comment]
I'm reassigning to @mstoodle to make sure this gets his attention. @charliegracie your suggestions are good and I will incorporate them when Mark confirms the wording.

**Comment by mstoodle** [comment]
A few comments.

The license header Angela included should apply to any code that has not been previously licensed. In other words, any previously unlicensed code contributed to OMR will become dual licensed under the Eclipse Public License v1.0 and Apache License 2.0 .

The contribution of any code with a preexisting license should ideally be discussed ahead of time via an issue tracker to avoid potential wasted effort or contributions that are simply incompatible with our project's licensing. Code that's licensed under a compatible license can be considered but we need to file a contributor questionnaire with the Eclipse foundation who will review the license and decide whether we are allowed to accept the contribution or not. 

I think we should be explicit that we cannot give anyone legal advice. The content and form of copyright lines are legal statements and so contributors are responsible for ensuring they have fulfilled their organization's requirements with respect to establishing or expressing (and appropriately updating) their organization's copyright on contributions.

Modifying an existing copyright line should only be performed by contributors from the organization holding the copyright. We encourage common sense and reason with respect to adding new copyright lines as we would prefer not to unnecessarily clutter our file headers with complex legal statements. For any questions, talk to a committer, ideally before contributing.

Sound reasonable?

**Comment by alin00** [comment]
I have updated according to your comments.

I left out the multiple-copyright owner case. That's now a "talk to a committer" scenario.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -28,11 +28,50 @@ Your contribution cannot be accepted unless you have an [Eclipse Foundation Cont
...
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    <First author> - initial implementation and documentation
+ *******************************************************************************/
+```
+* In the template, if you belong to the organization that owns the copyright,
```
Comment (mstoodle) [inline CONTRIBUTING.md]: I think this last statement should be removed as it sounds like advice to update a legal statement.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,11 +28,50 @@ Your contribution cannot be accepted unless you have an [Eclipse Foundation Cont
...
+* If you wish to contribute code under a different license, you must consult
+with a committer before contributing.
+* For any scenario not covered by this document, please discuss the copyright
+notice requirements with a committer before contributing.
+
+The template for the dual-license copyright notice is as follows:
+```c
+/*******************************************************************************
+ *
+ * (c) Copyright <copyright owner> <first year written>, <last year modified>
```
Comment (mstoodle) [inline CONTRIBUTING.md]: I think the copyright message should be abstracted to something like "Appropriate copyright message goes here if desired"

Comment (alin00) [inline CONTRIBUTING.md]: You mean I should delete the whole template?

Comment (mstoodle) [inline CONTRIBUTING.md]: Sorry, no: the copyright message is just the one line "(c) Copyright ..... "

The file header is mostly about license and only a little bit about copyright (that one line, actually).

File: CONTRIBUTING.md (hunk 1 above)
Comment (alin00) [inline CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 2 above)
Comment (alin00) [inline CONTRIBUTING.md]: Done.

**Comment by alin00** [comment]
Mark, I have amended the commit to address your comments.

**Comment by mstoodle** [comment]
Thanks, Angela. Looks good!

**Pull request merged**